src/sunadaptcontroller/noop/sunadaptcontroller_noop.c

```diff
@@ -0,0 +1,52 @@
+/* -----------------------------------------------------------------
+ * Programmer(s): Daniel R. Reynolds @ SMU
+ * -----------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2023, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------
+ * This is the implementation file for the SUNAdaptController_NOOP
+ * module.
+ * -----------------------------------------------------------------*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <sunadaptcontroller/sunadaptcontroller_noop.h>
+#include <sundials/sundials_math.h>
+
+
+/* -----------------------------------------------------------------
+ * exported functions
+ * ----------------------------------------------------------------- */
+
+/* -----------------------------------------------------------------
+ * Function to create a new No-op controller
+ */
+
+SUNAdaptController SUNAdaptController_NoOp(SUNContext sunctx)
+{
+  /* Create an empty controller object */
+  SUNAdaptController C = NULL;
```

Comment:
```suggestion
{
  SUNAdaptController C;
  
  if (sunctx == NULL) { return NULL; }
  
  /* Create an empty controller object */
  C = NULL;
```

Reply:
This is redundant, since `SUNAdaptController_NewEmpty` already performs this check.

Reply:
It is redundant but I don't think we should rely on a call to another function for input checks.